### ISSUE TITLE
Time range button: Screen reader is not announcing selected state for the currently selected date/month.

### ISSUE THREAD

**Issue by msftedad** [grafana/grafana#86653]
### What happened?

Screen reader does not announce the selected state for the currently selected month/day. It announces as `October 2022, button`. 
<img width="960" alt="code snippet-Screen reader is not announcing selected state for the currently selected date or month" src="https://github.com/grafana/grafana/assets/93735775/baadb770-1488-467c-a698-ff79afd7ef6f">

https://github.com/grafana/grafana/assets/93735775/635799d5-52e3-4ee9-8c17-7858d657600a

### What did you expect to happen?

The correct role as list item be defined for the dates, months, and year on the date picker, so that users can navigate on to these dates by arrow keys, and AT will announce state as selected for the selected date. 

Screen Reader announces `October 2022, selected 10 of 12`. 

### Did this work before?

No

### How do we reproduce it?

1. Open the URL [Grafana Play Home - Dashboards - Grafana](https://play.grafana.org/d/000000012/grafana-play-home?orgId=1) in edge Dev browser.
2. ‘Grafana Play Home’ page will open. 
3. Navigate to ‘Time range’ button & activate it. 
4. Navigate to ‘Date pickers’ present for ‘From’ & ‘To’ fields & activate it. 
5. Turn on narrator using ‘Ctrl+Windows+Enter’ keys. 
6. Navigate to selected date via tab key. 
7. Observe the screen reader narration. 

### Environment (with versions)?

Grafana: v11.1.0-69372 (51da96d94e)
OS: 24H2 (OS Build 26100.1) 
Browser: Edge version 125.0.2506.0 (Official build) dev (64-bit)

### Grafana platform?

None

**Comment by tskarhed** [comment]
This will most likely need an upstream contribution to [react-calendar](https://github.com/wojtekmaj/react-calendar/tree/main). The changes can take inspiration from [APG's Date Picker Dialog example](https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/examples/datepicker-dialog/).

There are issues in the react-calendar repo, which I think we can address and open a PR

- Aria-pressed https://github.com/wojtekmaj/react-calendar/issues/613
- Table, support for arrow key navigation https://github.com/wojtekmaj/react-calendar/issues/887
